lib/identity_cache/configuration_dsl.rb

```diff
@@ -227,7 +229,12 @@ def #{options[:cached_ids_name]}
           def #{options[:cached_accessor_name]}
             association_klass = association(:#{association}).klass
             if association_klass.should_use_cache? && !#{association}.loaded?
-              @#{options[:records_variable_name]} ||= #{options[:association_reflection].class_name}.fetch_multi(#{options[:cached_ids_name]})
+              @#{options[:records_variable_name]} ||= begin
+                #{options[:association_reflection].class_name}.fetch_multi(#{options[:cached_ids_name]}).map do |record|
+                  record.send('#{inverse_name}=', self) if #{inverse_name.present?}
```

Comment:
We don't want to set the association itself, since putting records loaded from the cache into an active record association would make it easier to unintentionally persist that data to the database, which could persist cache corruption.  Instead, we want to make sure they use the cached accessor to access it.  Can you re-use `set_inverse_of_cached_association` to do it the same way as embedded cached associations?